Renaming Microsoft.Framework.WebEncoders namespace

igor-susic opened the pull request:
Renamed `Microsoft.Framework.WebEncoders` namespace to `System.Text.Encodings.Web.Tests`

cc @ahsonkhan @GrabYourPitchforks 

Fix #40073

igor-susic committed:
Renaming the namespace Microsoft.Framework.WebEncoders to System.Text.Encodings.Web.Tests

- Namespace Microsoft.Framework.WebEncoders renamed to System.Text.Encodings.Web.Tests

igor-susic committed:
Changed aliases naming from camelCase to PascalCase

dnfclas commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/signed)](https://cla.dotnetfoundation.org/dotnet/corefx?pullRequest=40238) <br/>All CLA requirements met.

stephentoub commented on src/System.Text.Encodings.Web/tests/ConfigurableScalarTextEncoder.cs:
```diff
@@ -5,7 +5,7 @@
 using System;
 using System.Text.Encodings.Web;
```
This using statement should no longer be necessary. Same for similar cases in other files.

igor-susic replied:
@stephentoub No problem, it's my first contribution here and I didn't remember that in c# nesting namespaces can use everything from the layer above. Sorry :sweat_smile:

stephentoub commented on src/System.Text.Encodings.Web/tests/EncoderExtensionsTests.cs:
```diff
@@ -2,13 +2,14 @@
 
-using Microsoft.Framework.WebEncoders;
+using NewHtmlEncoder = System.Text.Encodings.Web.HtmlEncoder;
+using NewUrlEncoder = System.Text.Encodings.Web.UrlEncoder;
```
I'd very much prefer we not do this; it's inconsistent with how we write tests in general in corefx. And even if we did, the "New" part here doesn't make sense to me. Please revert this.

igor-susic replied:
@stephentoub Well, the namespace `System.Text.Encodings.Web.Tests` has file `TemporaryEncoderAdapters.cs` which defines `internal sealed class HtmlEncoder` and `internal sealed class UrlEncoder`, and test is testing classes from `System.Text.Encodings.Web`. Idea for those "aliases" I got from the `PerformanceTests.cs` in the same namespace. What would be the right way of solving this naming problem from the perspective of someone who has expertise like you? Just wondering, I would like to do it the right way and meet `corefx's` standard, that would make me more happy that just reverting, if that is an option :blush:

tarekgh replied:
The classes in file TemporaryEncoderAdapters.cs  are temporary and we'll remove it later. we can just have a different namespace in TemporaryEncoderAdapters.cs different than file TemporaryEncoderAdapters.cs 

---
In reply to: [313239124](https://github.com/dotnet/corefx/pull/40238#discussion_r313239124) [](ancestors = 313239124)

stephentoub reviewed (commented):
Thanks.

tarekgh commented:
other than Stephen's comments, LGTM.

igor-susic committed:
Deleted unnecessary using directives and reverted EncoderExtensionsTests

Deleted unnecessary using directives. Reverted changes made in EncoderCommonTests.cs to commit db1dd29edac07716ac27b8916d5e12f9a85b4978 and changed using directive for test namespace to right one.

igor-susic commented:
> other than Stephen's comments, LGTM.

Requested changes are implemented, I would appreciate if someone could take another look.

cc @tarekgh @stephentoub

ahsonkhan commented on src/System.Text.Encodings.Web/tests/EncoderExtensionsTests.cs:
```diff
@@ -2,7 +2,7 @@
 // See the LICENSE file in the project root for more information.
 
-using Microsoft.Framework.WebEncoders;
+using System.Text.Encodings.Web.Tests;
```
Ideally this would be within the `S.T.Encodings.Web.Tests` namespace as well and we wouldn't need this using, but given we have the custom adapter encoders, it makes sense to keep this as is for now (and remove the temporary adapters in the future).

ahsonkhan reviewed (approved):
LGTM

tarekgh commented:
Thanks @igor-susic
